src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -110,32 +107,74 @@ internal async Task SearchAsync(CancellationToken cancellationToken)
+            var searchGeneratedDocuments = isFullyLoaded;
 
-            // If we're fully loaded then we're done at this point.  All the searches would have been against the latest
-            // computed data and we don't need to do anything else.
-            if (isFullyLoaded)
-                return;
+            var projectCount = orderedProjects.Sum(g => g.Length);
 
-            // We weren't fully loaded *but* we reported some items to the user, then consider that good enough for now.
-            // The user will have some results they can use, and (in the case that we actually examined the cache for
-            // data) we will tell the user that the results may be incomplete/inaccurate and they should try again soon.
-            if (foundItems)
-                return;
+            if (!isFullyLoaded)
+            {
+                // If we're not fully loaded, do a quick search through any cached data we have.
+                // If we don't find any results, load all the projects and search through them.
+                await _progress.AddItemsAsync(projectCount * 2, cancellationToken).ConfigureAwait(false);
 
-            // We didn't have any items reported *and* we weren't fully loaded.  Try searching the projects again, but this
-            // time tell them to use the latest data.  The ensures the user may get some result instead of nothing.
-            var foundFullItems = await ProcessProjectsAsync(orderedProjects, isFullyLoaded: true, cancellationToken).ConfigureAwait(false);
+                await SearchCachedDocumentsAsync(orderedProjects, seenItems, cancellationToken).ConfigureAwait(false);
 
-            // Report a telemetry even to track if we found uncached items after failing to find cached items.
-            // In practice if we see that we are always finding uncached items, then it's likely something
-            // has broken in the caching system since we would expect to normally find values there.  Specifically
-            // we expect: foundFullItems <<< not foundFullItems.
+                // If searching cached data returned any results, then we're done.  We've at least shown some results
+                // to the user.  That will hopefully serve them well enough until the solution fully loads.
+                if (seenItems.Count > 0)
+                {
+                    // Ensure that we actually complete all our workitems so that the progress bar completes.
+                    await ProcessOrderedProjectsAsync(orderedProjects, seenItems,
+                        async (_, _, _) => await _progress.ItemCompletedAsync(cancellationToken).ConfigureAwait(false),
+                        cancellationToken).ConfigureAwait(false);
+                    return;
+                }
+
+                await SearchFullyLoadedProjectsAsync(orderedProjects, seenItems, cancellationToken).ConfigureAwait(false);
```

Comment:
I don't understand why this isn't searching generated documents? The solution wasn't fully loaded, so we searched the cache and didn't find anything, so we're gonna wait for the projects to load and search them.. why not also search the generated documents?

Reply:
That's a good point :-)